### PR TITLE
add configuration to determine number of blocks to search last transactions

### PR DESCRIPTION
**Pull request by hugo-dc**
retrieving transactions in last 128 blocks takes too much time in aleth.

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -20,7 +22,11 @@ router.get('/', function(req, res, next) {
     }, function(lastBlock, callback) {
       var blocks = [];
 
-      var blockCount = 128;
+      nodeVersion = req.app.locals.nodeStatus.version;
```
Comment (axic) [inline routes/index.js]: If this is queried every single time, then this value should be cached at start up time.

File: routes/index.js [hunk 2]
```diff
@@ -4,6 +4,8 @@ var router = express.Router();
 var async = require('async');
 var Web3 = require('web3');
 var web3complete = require('web3-complete');
+var nodeVersion = '';
+var blockCount = 128;
```
Comment (axic) [inline routes/index.js]: I think maybe the better approach is moving this into the configuration file. Then it is up to the person running it to decide what is a good setting.

Comment (hugo-dc) [reply routes/index.js]: I've added it as a configuration, this config will also be used when showing an account in order do determine how many blocks too look into to get last transactions for the account.

File: config.js [hunk 3]
```diff
@@ -19,6 +19,10 @@ var config = function (endpoint) {
   }
   this.faucetAddress = "0x9fa4f23079bae7a7a5c392b67e18093310315bfc";
   //this.privateKey = new Buffer('<insert_private_key_here>', 'hex');
+
+  // look for transactions the last N blocks
+  // recommended: aleth=10, geth=128
+  this.blockCount = 10;
```
Comment (axic) [inline config.js]: Can you keep this at 128 ?

Comment (hugo-dc) [reply config.js]: changed to 128, and added a note in README.

File: config.py [hunk 4]
```diff
@@ -8,17 +8,21 @@
                             help='faucet address')
 parser.add_argument('--private-key', type=str, nargs='?',
                             help='faucet private key')
+parser.add_argument('--block-count', type=int, nargs='?',
```
Comment (axic) [inline config.py:11]: Changed this to `int`.

File: config.py [hunk 5]
```diff
@@ -8,17 +8,21 @@
...
 template = parse_args.template
 faucet_address = parse_args.faucet_address
 private_key = parse_args.private_key
+block_count = parse_args.block_count
 
 try:
   with open(template) as f:
     template = f.read().replace('{privateKey}', private_key)
     template = template.replace('{faucetAddress}', faucet_address)
+    template = template.replace('{blockCount}', str(block_count))
```
Comment (hugo-dc) [inline config.py:25]: `replace` requires string arguments.

File: config.py (hunk 4 above)
Comment (hugo-dc) [reply config.py:11]: Looks good, I converted `block_count` to string (see comment below)

Comment (axic) [reply config.py:11]: Thanks!

**Review by axic — APPROVED** [review]

**Pull request merged**